c2c/template.py

```diff
@@ -49,6 +51,32 @@ def check_output(cmd, cwd=None, stdin=None, stderr=None, shell=False):  # noqa
         return out
 
 
+def split_no_escapes(string, char):
+    sections = string.split(char)
+    sections = [i + (char if i[-1] == "\\" else "") for i in sections]
+    result = ["" for i in sections]
+    j = 0
+    for s in sections:
+        result[j] += s
+        j += (1 if s[-1] != char else 0)
```

Comment:
Up to there seems overly complicated. If you are doing what I think you are doing, this can be done with a single regex with the magic of a negative look behind:
DOT_SPLITTER = re.compile(r"(?<!\\)\.")
...
result = DOT_SPLITTER.split(string)